spec/inputs/tcp_spec.rb

```diff
@@ -21,19 +21,27 @@
 #Cabin::Channel.get(LogStash).level = :debug
 describe LogStash::Inputs::Tcp, :ecs_compatibility_support do
 
-  def get_port
-    begin
-      # Start high to better avoid common services
-      port = rand(10000..65535)
-      s = TCPServer.new("127.0.0.1", port)
-      s.close
-      return port
-    rescue Errno::EADDRINUSE
-      retry
-    end
+  ##
+  # yield the block with a port that is available
+  # @return [Integer]: a port that is available
+  def find_available_port
+    with_bound_port(&:itself)
+  end
+
+  ##
+  # Yields block with a port that is unavailable
+  # @yieldparam port [Integer]
+  # @yieldreturn [Object]
+  # @return [Object]
+  def with_bound_port(port=0, &block)
+    server = TCPServer.new("::", port)
```

Comment:
tests used to focus on ipv4 and while I'm sure we also want to move tests to ipv6 not sure if this PR is the right place to do it.